services/packages/cleanup/cleanup.go

```diff
@@ -33,13 +33,7 @@ func CleanupTask(ctx context.Context, olderThan time.Duration) error {
 }
 
 func ExecuteCleanupRules(outerCtx context.Context) error {
-	ctx, committer, err := db.TxContext(outerCtx)
-	if err != nil {
-		return err
-	}
-	defer committer.Close()
-
-	err = packages_model.IterateEnabledCleanupRules(ctx, func(ctx context.Context, pcr *packages_model.PackageCleanupRule) error {
+	return packages_model.IterateEnabledCleanupRules(outerCtx, func(ctx context.Context, pcr *packages_model.PackageCleanupRule) error {
```

Comment:
Why the `db.TxContext`? `outerCtx` seems to be a bug

Reply:
Removing the transaction allows partial cleanup to succeed. In the previous implementation, the entire cleanup process would either fully succeed or completely fail.

Reply:
https://github.com/go-gitea/gitea/pull/34632/commits/fb79731d957e4d4004e1b36be5189defd8397885